py/picca/pk1d/postproc_pk1d.py

```diff
@@ -0,0 +1,331 @@
+
+            index = (nbins_k * izbin) + ikbin # index to be filled in table
+            meanP1D_table['zbin'][index] = zbin_centers[izbin]
+            meanP1D_table['index_zbin'][index] = izbin
+
+            N = np.ma.count(p1d_table['k'][select]) # Counts the number of chunks in each (z,k) bin
+            meanP1D_table['N'][index] = N
+
+            for ic, c in enumerate(p1d_table_cols):
+
+                if N==0:
+                    userprint('Warning: 0 chunks found in bin '+str(zbin_edges[izbin])+'<z<'+str(zbin_edges[izbin+1])+
+                          ', '+str(kbin_edges[ikbin])+'<k<'+str(kbin_edges[ikbin+1]))
+                    for stats in stats_array:
+                        meanP1D_table[stats+c][index] = np.nan
+                    continue
+
+                if weight_method=='fit_snr':
+                    snr_bin_edges = np.arange(1,11,1)  # 9 SNR bins are used for the fit
+                    snr_bins = (snr_bin_edges[:-1]+snr_bin_edges[1:])/2
+                    def variance_function(snr, a, b):
+                        return (a/(snr-1)**2) + b
+                    data_values = p1d_table[c][select]
+                    data_snr = p1d_table['forest_snr'][select]
+                    # Fit function to observed dispersion:
+                    standard_dev,_,_ = binned_statistic(data_snr, data_values,
+                                                                 statistic='std', bins=snr_bin_edges)
+                    coef, coef_cov = curve_fit(variance_function, snr_bins, standard_dev**2, bounds=(0,np.inf))
+                    # Model variance from fit function
+                    data_snr[data_snr>11] = 11
+                    data_snr[ data_snr<1.01] = 1.01
+                    variance_estimated = variance_function(data_snr, *coef)
+                    weights = 1. / variance_estimated
+                    mean = np.average((p1d_table[c][select]), weights=weights)
+                    error = np.sqrt(1. / np.sum(weights))
+                    if output_snrfit is not None and c=='Pk':
+                        snrfit_table[index, 0:4] = [zbin_centers[izbin], (kbin+kbin_edges[ikbin+1])/2.,
+                                                    coef[0], coef[1]]
+                        snrfit_table[index, 4:] = standard_dev
+
+                elif weight_method=='simple_snr':
+                    # for forests with snr>snr_limit (hardcoded to 4 as of now), 
+                    # the weight is fixed to (snr_limit - 1)**2 = 9
+                    snr_limit = 4 
+                    forest_snr = p1d_table['forest_snr'][select]
+                    w, = np.where(forest_snr <= 1)
+                    if len(w)>0: raise RuntimeError('Cannot add weights with SNR<=1.')
+                    weights = (forest_snr - 1)**2
```

Comment:
ok, I guess the snr-1 is the cause for the error message above, but where does it come from? why not just weight with snr**2

Reply:
My question would be if that cut is really needed, we are using it for continuum fitting with constant weights as without a cut and with constant weights you'd be dominated by the noisier  spectra and the continuum fit becomes bad.
But in principle one could hope that moving to ivar weights for the continuum fitting and the avarages would be possible and one could use also spectra with lower SNR. In which case SNR-1 wouldn't work.
I'm also somewhat surprised that should work better...

Reply:
- I agree we should try to use SNR<1 data for Y1, and in that case the weighting scheme would change.
- The SNR-1 is really empirical, don't know exactly why it works better. However, "SNR" is not *the* optimal variable to weight forest's individual P(k)s, since the "signal" in P(k)s is from the flux fluctuations and not the flux itself. So I dont expect the optimal weight = SNR**2 either on a theoretical ground. We just use it as a simple forest-base proxy, since it's there.

Reply:
but we rescale the noise to be the noise fluctuations, so the SNR is unchanged in the process, compared to the original one... But yes, not sure it's the optimal weight either. Let's leave as is for now and study further at a later time...